pkg/service/utgen/ai.go

```diff
@@ -107,6 +109,26 @@ func (ai *AIClient) Call(ctx context.Context, prompt *Prompt, maxTokens int) (st
 
 	var apiBaseURL string
 
+	// Signal handling for Ctrl+C (SIGINT)
+	sigChan := make(chan os.Signal, 1)
+	signal.Notify(sigChan, os.Interrupt, syscall.SIGTERM)
+	defer signal.Stop(sigChan)
+
+	// Create a cancellable context for the request
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	// Start a goroutine to listen for Ctrl+C and cancel the context
+	go func() {
+		select {
+		case <-sigChan:
+			fmt.Println("Received interrupt signal. Cancelling AI request...")
+			cancel() // Cancel the context if Ctrl+C is pressed
```

Comment:
are we sure we want to print this log?for other use cases do we print?

Reply:
Not necessary cause Ctrl+c we have a log, maybe we can omit that. your take?

Reply:
I don't generally prefer print statements and we've better logger at place. so we can omit this